src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,72 @@
+package parser;
+
+import java.util.Scanner;
+
+import command.AddDeadlineCommand;
+import command.AddEventCommand;
+import command.AddTodoCommand;
+import command.Command;
+import command.DeleteCommand;
+import command.EmptyCommand;
+import command.ExitCommand;
+import command.FindCommand;
+import command.InvalidCommand;
+import command.ListCommand;
+import command.MarkCommand;
+import command.UnmarkCommand;
+import task.TaskList;
+import ui.Ui;
+
+/**
+ * <h1>Parser class</h1>
+ * Parses the user's input and generates the appropriate command.
+ */
+public class Parser {
+    private Scanner sc;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Creates a Parser object.
+     *
+     * @param tasks list of Tasks the user currently has.
+     * @param ui the User Interface that prints out the output.
+     */
+    public Parser(TaskList tasks, Ui ui) {
+        this.sc = new Scanner(System.in);
+        this.tasks = tasks;
+        this.ui = ui;
+    }
+
+    /**
+     * Returns the appropriate Command after parsing the input String.
+     *
+     * @param input String to be parsed.
+     * @return Command to be executed.
+     */
+    public Command parse(String input) {
```

Comment:
Some suggestions: maybe you can use switch statement for better organization : )